scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala

```diff
@@ -39,9 +39,9 @@ object CumulativeSum {
     def cumulativeSum(
       implicit sg: Semigroup[V], ordU: Ordering[U], ordK: Ordering[K]): SortedGrouped[K, (U, V)] = {
       pipe.group
-        .sortBy { case (u: U, _) => u }
+        .sortBy { case (u, _) => u }
```

Comment:
can it tell enough to know the "shape" is `(k, (u, v))`?

This is one thing I'm often confused about with case matching in scala, how to make sure you aren't accidentally doing reflection in a case match in a way that will never even match.